packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/util/ClientModelUtil.java

```diff
@@ -447,8 +448,19 @@ public static ClientModel getClientModel(String name) {
     public static boolean isClientModel(IType type) {
         if (type instanceof ClassType) {
             ClassType classType = (ClassType) type;
-            return classType.getPackage().startsWith(JavaSettings.getInstance().getPackage())
-                && getClientModel(classType.getName()) != null;
+            /*
+             * It is possible in TypeSpec that the models be in different package of the package in JavaSettings.
+             * Therefore, we now check:
+             * 1. The package of the class is not in java.*
+             * 2. The package of the class is not in core package (clientcore or azure-core)
+             */
+            // final boolean typeInExternalPackage =
+            // !classType.getPackage().startsWith(JavaSettings.getInstance().getPackage());
+            final boolean typeInExternalPackage = classType.getPackage().startsWith("java.")
+                || classType.getPackage().startsWith(ExternalPackage.CLIENTCORE_PACKAGE_NAME)
+                || classType.getPackage().startsWith(ExternalPackage.AZURE_CORE_PACKAGE_NAME)
+                || classType.getPackage().startsWith("com.azure.v2.core");
```

Comment:
Welcome advise on it.

One shortcoming of existing logic (changing that could take quite some efforts) is that Model/Class is cached by its name, not the full name.

Reply:
> (wondering whether spotless can postprocess and simplify name reference -


this showed up in search - there is a plugin `cleanthat` which has a [config](https://github.com/solven-eu/cleanthat/blob/master/java/src/main/java/eu/solven/cleanthat/engine/java/refactorer/mutators/ImportQualifiedTokens.java), as per doc it is intended to simplify the fully qualified names

Reply:
Thanks. Will try it when the request arise that we must support models from different packages (TypeSpec can already write this, but so far we didn't see such case).